packages/components/bin/build-metadata.mts

```diff
@@ -0,0 +1,76 @@
+import { mkdir, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { parse } from 'react-docgen-typescript';
+
+type ComponentMetadata = {
+	props: Record<
+		string,
+		{
+			description: string;
+			defaultValue?: { value: any };
+			required: boolean;
+			type: { name: string };
+		}
+	>;
+};
+
+const DOCUMENTED_COMPONENTS = [ 'breadcrumbs', 'summary-button' ];
+
+const files = DOCUMENTED_COMPONENTS.map( ( component ) => `${ component }/index.tsx` );
+
+const pick = < O extends Record< string, any >, K extends Array< keyof O > >(
+	obj: O,
+	keys: K
+): Pick< O, K[ number ] > =>
+	Object.fromEntries( Object.entries( obj ).filter( ( [ key ] ) => keys.includes( key ) ) ) as Pick<
+		O,
+		K[ number ]
+	>;
+
+const omit = < O extends Record< string, any >, K extends Array< keyof O > >(
+	obj: O,
+	keys: K
+): Omit< O, K[ number ] > =>
+	Object.fromEntries(
+		Object.entries( obj ).filter( ( [ key ] ) => ! keys.includes( key ) )
+	) as Omit< O, K[ number ] >;
+
+const mapValues = < V, MV >(
+	obj: Record< string, V >,
+	fn: ( value: V ) => MV
+): Record< string, MV > =>
+	Object.fromEntries( Object.entries( obj ).map( ( [ key, value ] ) => [ key, fn( value ) ] ) );
+
+async function getMetadataEntry( file: string ): Promise< [ string, ComponentMetadata ] > {
+	const parsed = await parse( join( process.cwd(), 'src', file ) );
+	const [ { displayName, props } ] = parsed;
+	return [
+		displayName,
+		{
+			props: mapValues( omit( props, [ 'ref', 'key' ] ), ( value ) =>
+				pick( value, [ 'description', 'defaultValue', 'required', 'type' ] )
+			),
+		},
+	];
+}
+
+const metadata = Object.fromEntries( await Promise.all( files.map( getMetadataEntry ) ) );
+
+await mkdir( 'dist/types', { recursive: true } );
+
+await Promise.all( [
+	writeFile( 'dist/metadata.js', `export default ${ JSON.stringify( metadata, null, 2 ) };` ),
```

Comment:
I'm thinking what the workflow would be like on PRs that make docs-affecting changes to the components package. Ideally, we'd prompt two checks as part of the code review:

1. Did prop types change in the way you expected? (When dealing with types composed from other types using `Pick`/`Omit`, you sometimes fail to notice that your type changes affected the prop types of other components in ways you didn't intend.)
2. Are your TSDocs rendering correctly on the site? (Badly formatted markdown, unsupported `@tags`, etc)

What do you think? Do a diff of `dist/metadata.js` on CI, and post a comment with live links to the doc site?